### PR TITLE
Werners suggestions for advanced search

### PR DESCRIPTION
**Pull request by baerlach-git**
implementing most of werner feedback concerning advanced search:

- added a "reset filter button"
- added a notification that reminds users to apply their search filters if the filter configuration does not match the currently active search filter
- made it so text fields are emptied if users select the "Nichts ausgewählt" filter option
- fixed a bug where most filters only worked properly if filter terms where put in quotation marks
- removed filter for collections
- hid the description filter and face-tag filter from users that are not logged in
- made it possible to only search within comments

### CODE REVIEW

**Review by rheaSPK — APPROVED** [review]
This is an alibi approve ;)

File: projects/bp-gallery/src/components/views/search/SearchView.tsx [hunk 1]
```diff
@@ -63,16 +63,24 @@ const SearchView = () => {
     };
   }, [searchParams]);
 
+  console.log(getSearchResultHits(queryParams, filter, textFilter, searchIndex));
```
Comment (MariusDoe) [inline projects/bp-gallery/src/components/views/search/SearchView.tsx:66]: Maybe remove that?

File: projects/bp-gallery/src/components/views/search/SearchView.tsx [hunk 2]
```diff
@@ -63,16 +63,24 @@ const SearchView = () => {
...
+            : ''
+        )
+        .filter(id => id !== ''),
+    [queryParams, textFilter, filter, searchIndex]
   );
-  if (import.meta.env.MODE === 'development') {
-    console.log(searchResultIds);
-  }
+
+  console.log('resultIds:', searchResultIds);
```
Comment (MariusDoe) [inline projects/bp-gallery/src/components/views/search/SearchView.tsx:82]: same

**Review by MariusDoe — CHANGES_REQUESTED** [review]

File: projects/bp-gallery/src/components/views/search/SearchView.tsx (hunk 1 above)
Comment (baerlach-git) [reply projects/bp-gallery/src/components/views/search/SearchView.tsx:66]: true

**Review by MariusDoe — APPROVED** [review]

**Pull request merged**